mmv1/third_party/terraform/services/compute/resource_compute_region_instance_group_manager_test.go.tmpl

```diff
@@ -1925,6 +1969,174 @@ resource "google_compute_region_instance_group_manager" "sr-igm" {
 }
 {{- end }}
 
+{{ if ne $.TargetVersionName `ga` -}}
+func testAccRegionInstanceGroupManager_instanceFlexibilityPolicy(network, template, igm string) string {
+	return fmt.Sprintf(`
+data "google_compute_image" "my_image" {
+	family  = "debian-11"
+	project = "debian-cloud"
+}
+
+resource "google_compute_network" "igm-basic" {
+	name = "%s"
+}
+
+resource "google_compute_instance_template" "igm-basic" {
+	name           = "%s"
+	machine_type   = "e2-medium"
+	disk {
+		source_image = data.google_compute_image.my_image.self_link
+		auto_delete  = true
+		boot         = true
+		device_name  = "stateful-disk"
+	}
+	network_interface {
+		network = "default"
+	}
+}
+
+resource "google_compute_region_instance_group_manager" "igm-basic" {
+	description = "Terraform test instance group manager"
+	name        = "%s"
+
+	version {
+		instance_template = google_compute_instance_template.igm-basic.self_link
+		name              = "primary"
+	}
+
+	base_instance_name        = "tf-test-igm-basic"
+	region                    = "us-central1"
+	target_size               = 2
+	distribution_policy_target_shape = "ANY_SINGLE_ZONE"
+	update_policy {
+		instance_redistribution_type = "NONE"
+		type                         = "OPPORTUNISTIC"
+		minimal_action               = "REPLACE"
+		max_surge_fixed              = 0
+		max_unavailable_fixed        = 6
+	}
+	instance_flexibility_policy {
+		instance_selections {
+			name = "instance_selection_name_1"
+			rank = 2
+			machine_types = ["n1-standard-16"]
+		}
+		instance_selections {
+			name = "instance_selection_name_2"
+			rank = 1
```

Comment:
What happens if we omit `rank` / can you include a configuration that does?

Reply:
AFAIK, we can either set rank for all the instanceSelections or none. On the update example, I remove the rank from that instanceSelection. And since it is optional, it pics the changes accordingly. 
https://screenshot.googleplex.com/ApGbuisMpvaWFVQ